sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -140,6 +140,17 @@ object IntegerLiteral {
 }
 
 /**
+  * Extractor for retrieving Boolean literals.
+  */
+object BooleanLiteral {
```

Comment:
This looks weird. Called BooleanLiteral but actually it matches literals of IntegerType?